src/module_event_handlers.c

```diff
@@ -262,6 +273,8 @@ static void RG_AfterForkChild() {
 }
 
 static void _RegisterForkHooks() {
+	redis_main_thread_id = pthread_self();  // This function is being called on the main thread context.
```

Comment:
@OfirMos is this called from the main thread?

Reply:
Yes this is called from RedisModule_Load() which is being execute on the main thread context.